Sync with CWL-Airflow 1.2.N

heylel-b-sh opened the pull request:
Should work with CWL-Airflow 1.2.N

Applied changes:
- refactored `post_build.sh` script. However, lines starting from **_138_** remained unchanged as their usage is not absolutely clear and cannot be properly refactored at the moment
- In `SatelliteApp.ts`
   - all CWL-Airflow related executables should be run from `bin_portable` folder
   - no need to copy `clean_dag_run.py` as it's created by `cwl-airflow init --upgrade` command, where `--upgrade` will additionally convert all DAGs to the new format
   - no need to call `airflow initdb/upgradedb` as `cwl-airflow init` does it
   - `process_report` connection doesn't require `--extra` parameter anymore
   - no need to set `PYTHONPATH` as all CWL-Airflow related executables are bash scripts that already set all required environment variables

heylel-b-sh committed:
Update post_build script and CWL-Airflow related paths

Not tested

heylel-b-sh committed:
Update biowardrobe-ng version to 2.0.0

heylel-b-sh committed:
Update CWL-Airflow version in build script. Fix issues in SatelliteApp.ts

heylel-b-sh committed:
Not important changes

heylel-b-sh committed:
Need to remove airflow connection before it can be updated

heylel-b-sh committed:
Fixed bug in airflowInit and airflowUpdate functions.

No need to create dag folder and copy there clean_dag_run.py
anymore as it's done by cwl-airflow init command. Although,
we still need to delete the original clean_dag_run.py otherwise
it won't be updated.
We also need to make sure the process_report connection is
updated. For that we remove and add it again. All command run
with runCommands should be executed sequentially, therefore
theay are combined with &&

heylel-b-sh committed:
Use CWL-Airflow Python 3.8 version

heylel-b-sh committed:
Minor correction, not important

heylel-b-sh committed:
Downgrade to Node 12.16.3

portah commented on electron/SatelliteApp.ts:
```diff
@@ -436,16 +436,16 @@ export class SatelliteApp {
+        };
         if (this.proxySettings) {
-            env_var["https_proxy"] = `${this.proxySettings}`
+            env_var['https_proxy'] = `${this.proxySettings}`;
```
As we discussed ```proxy``` should be part of ```this.satelliteSettings```. So the code should be ```this.satelliteSettings.proxy```

heylel-b-sh replied:
Done

portah commented on electron/SatelliteApp.ts:
```diff
@@ -436,16 +436,16 @@ export class SatelliteApp {
-        }
+        };
         if (this.proxySettings) {
-            env_var["https_proxy"] = `${this.proxySettings}`
```
https & http

heylel-b-sh replied:
Done

portah commented on electron/SatelliteApp.ts:
```diff
@@ -615,9 +615,16 @@ export class SatelliteApp {
+        }
 
         const init_commands = [
-            `cwl-airflow init`
```
Do we need to replace init with init --update ? It's the first start no anything

heylel-b-sh replied:
If there is no any DAGs in the dags folder `--upgrade` won't do anything, so you can leave just `cwl-airflow init`

portah commented on electron/SatelliteApp.ts:
```diff
@@ -627,7 +634,7 @@ export class SatelliteApp {
         await fs.promises.writeFile(`${this.airflowSettings.AIRFLOW_HOME}/airflow.cfg`, stringify(airflowConfig, { whitespace: true }));
-        // await fs.promises.mkdir(`${this.airflowSettings.AIRFLOW_HOME}/dags`, { recursive: true });
+        await fs.promises.mkdir(`${this.airflowSettings.AIRFLOW_HOME}/dags`, { recursive: true });
         // await fs.promises.copyFile(`${this.airflow_base_path}/Resources/app/cwl_airflow/dags/clean_dag_run.py`, `${this.airflowSettings.AIRFLOW_HOME}/dags/clean_dag_run.py`);
```
if we run just init with now --upgrade do we still need to copy clean dag?

heylel-b-sh replied:
It's not present in the latest commit

portah commented on electron/SatelliteApp.ts:
```diff
@@ -252,8 +252,7 @@ export class SatelliteApp {
                 PM2_API_PORT: this.satelliteSettings.pm2Port,
                 // PM2_HOME: this.pm2_home,
```
maybe PM2 needs proxy too? it reports to home a cool staff of data

heylel-b-sh replied:
I would prefer to add it later when needed, but it's up to you

heylel-b-sh committed:
Move proxy setting to satelliteSettings object

For node add  both https and http

heylel-b-sh committed:
Set max_active_runs_per_dag to 1

Should prevent starting the new DAG-run if the previous
DAG-run of the same DAG failed with retry
